base/applications/charmap/map.c

```diff
@@ -312,64 +313,53 @@ OnClick(PMAP infoPtr,
 
-    pt.x = ptx;
-    pt.y = pty;
+    x = ptx / infoPtr->CellSize.cx;
+    y = pty / infoPtr->CellSize.cy;
 
-    for (x = 0; x < XCELLS; x++)
-    for (y = 0; y < YCELLS; y++)
+    /* if the cell is not already active */
+    if (!infoPtr->Cells[y][x].bActive)
     {
-        if (PtInRect(&infoPtr->Cells[y][x].CellInt,
-                     pt))
+        /* set previous active cell to inactive */
+        if (infoPtr->pActiveCell)
         {
-            /* if the cell is not already active */
-            if (!infoPtr->Cells[y][x].bActive)
-            {
-                /* set previous active cell to inactive */
-                if (infoPtr->pActiveCell)
-                {
-                    /* invalidate normal cells, required when
-                     * moving a small active cell via keyboard */
-                    if (!infoPtr->pActiveCell->bLarge)
-                    {
-                        InvalidateRect(infoPtr->hMapWnd,
-                                       &infoPtr->pActiveCell->CellInt,
-                                       TRUE);
-                    }
-
-                    infoPtr->pActiveCell->bActive = FALSE;
-                    infoPtr->pActiveCell->bLarge = FALSE;
-                }
-
-                /* set new cell to active */
-                infoPtr->pActiveCell = &infoPtr->Cells[y][x];
-                infoPtr->pActiveCell->bActive = TRUE;
-                infoPtr->pActiveCell->bLarge = TRUE;
-                if (infoPtr->hLrgWnd)
-                    MoveLargeCell(infoPtr);
-                else
-                    CreateLargeCell(infoPtr);
-            }
-            else
+            /* invalidate normal cells, required when
+                * moving a small active cell via keyboard */
+            if (!infoPtr->pActiveCell->bLarge)
             {
-                /* flick between large and small */
-                if (infoPtr->pActiveCell->bLarge)
-                {
-                    DestroyWindow(infoPtr->hLrgWnd);
-                    infoPtr->hLrgWnd = NULL;
-                }
-                else
-                {
-                    CreateLargeCell(infoPtr);
-                }
-
-                infoPtr->pActiveCell->bLarge = (infoPtr->pActiveCell->bLarge) ? FALSE : TRUE;
+                InvalidateRect(infoPtr->hMapWnd,
+                                &infoPtr->pActiveCell->CellInt,
+                                TRUE);
             }
 
-            break;
+            infoPtr->pActiveCell->bActive = FALSE;
+            infoPtr->pActiveCell->bLarge = FALSE;
         }
+
+        /* set new cell to active */
+        infoPtr->pActiveCell = &infoPtr->Cells[y][x];
+        infoPtr->pActiveCell->bActive = TRUE;
+        infoPtr->pActiveCell->bLarge = TRUE;
+        if (infoPtr->hLrgWnd)
+            MoveLargeCell(infoPtr);
+        else
+            CreateLargeCell(infoPtr);
+    }
+    else
+    {
+        /* flick between large and small */
+        /*if (infoPtr->pActiveCell->bLarge)
```

Comment:
If this is no longer needed, please remove it. If there's some reason that commenting it out is better, there should be a comment explaining why.

Reply:
Sounds good, I've removed this code